### PR TITLE
ImageProcessingBlock ported to WebGPU support

### PR DESCRIPTION
**Pull request by deltakosh**

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

File: packages/dev/core/src/Materials/Node/Blocks/Fragment/imageProcessingBlock.ts [hunk 1]
```diff
@@ -156,16 +156,16 @@ export class ImageProcessingBlock extends NodeMaterialBlock {
             if (color.connectedPoint!.type === NodeMaterialBlockConnectionPointTypes.Color4 || color.connectedPoint!.type === NodeMaterialBlockConnectionPointTypes.Vector4) {
                 state.compilationString += `${state._declareOutput(output)} = ${color.associatedVariableName};\n`;
             } else {
-                state.compilationString += `${state._declareOutput(output)} = vec4(${color.associatedVariableName}, 1.0);\n`;
+                state.compilationString += `${state._declareOutput(output)} = vec4${state.fSuffix}(${color.associatedVariableName}, 1.0);\n`;
             }
             state.compilationString += `#ifdef IMAGEPROCESSINGPOSTPROCESS\n`;
             if (this.convertInputToLinearSpace) {
-                state.compilationString += `${output.associatedVariableName}.rgb = toLinearSpace(${color.associatedVariableName}.rgb);\n`;
+                state.compilationString += `${output.associatedVariableName} = vec4${state.fSuffix}(toLinearSpaceVec3(${color.associatedVariableName}.rgb), ${color.associatedVariableName}.a);\n`;
```
Comment (Popov72) [inline packages/dev/core/src/Materials/Node/Blocks/Fragment/imageProcessingBlock.ts:163]: Don't we also need a `toLinearSpaceVec3` in WebGL? It seems this function only exists in `helperFunctions.fx` for WGSL, not for WebGL.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/15188/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15188/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15188/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/15188/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/15188/merge
https://gui.babylonjs.com/?snapshot=refs/pull/15188/merge
https://nme.babylonjs.com/?snapshot=refs/pull/15188/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/15188/merge#BCU1XR#0

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15188/merge/testResults/webgpuplaywright/index.html

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15188/merge/testResults/webgl2playwright/index.html

File: packages/dev/core/src/Materials/Node/Blocks/Fragment/imageProcessingBlock.ts (hunk 1 above)
Comment (deltakosh) [reply packages/dev/core/src/Materials/Node/Blocks/Fragment/imageProcessingBlock.ts:163]: nice catch!!

**Review by Popov72 — APPROVED** [review]

**Pull request merged**